test/run/test_stages.py

```diff
@@ -452,6 +458,62 @@ def test_sgdisk(self):
             for p in table["partitions"]:
                 p["node"] = os.path.basename(p["node"])
 
+            # Old versions of sfdisk (e.g. on RHEL-8), do not include
+            # the 'sectorsize' in the output, so we delete it from the
+            # expected output if it is not present in the actual output
+            if "sectorsize" not in table:
+                del want["partitiontable"]["sectorsize"]
+
+            self.assertEqual(have, want)
+
+            # cache the downloaded data for the files source
+            osb.copy_source_data(self.store, "org.osbuild.files")
+
+    @unittest.skipUnless(have_sfdisk_with_json(), "Need sfdisk with JSON support")
+    def test_sfdisk(self):
+        datadir = self.locate_test_data()
+        testdir = os.path.join(datadir, "stages", "sfdisk")
+
+        imgname = "disk.img"
+
+        with open(os.path.join(testdir, f"{imgname}.json"), "r", encoding="utf8") as f:
+            want = json.load(f)
+
+        with self.osbuild as osb, tempfile.TemporaryDirectory(dir="/var/tmp") as outdir:
+
+            osb.compile_file(os.path.join(testdir, "sfdisk.json"),
+                             checkpoints=["tree"],
+                             exports=["tree"],
+                             output_dir=outdir)
+
+            target = os.path.join(outdir, "tree", imgname)
+
+            assert os.path.exists(target)
+
+            r = subprocess.run(["sfdisk", "--json", target],
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               encoding="utf8",
+                               check=False)
+
+            have = json.loads(r.stdout)
+
+            table = have["partitiontable"]
+
+            # remove entries that are not stable across `sfdisk`
```

Comment:
I'd be tempted to turn this block into some kind of table cleanup function so it doesn't need to be repeated, and gives your one place to add new things when they come up in the future :)

Reply:
I had the same temptation (even a version of it), but then didn't want to over-engineer the solution. But since you suggested it, I'll blame the over-engineering on your solution 😅 

I added a new commit that refactors the code.